### PR TITLE
feat: Add proper INodeTypes polyfill for Task Runner (no-changelog)

### PR DESCRIPTION
**Pull request by valya**
## Summary

Some WorkflowDataProxy code paths end up looking up node types. This polyfill is based on the frontend's approach, so should be enough to get it working.

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/task-runner/src/node-types.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	INodeTypes,
+	IVersionedNodeType,
+} from 'n8n-workflow';
+
+type VersionedTypes = Map<number, INodeTypeDescription>;
+
+export const DEFAULT_NODETYPE_VERSION = 1;
+
+export class TaskRunnerNodeTypes implements INodeTypes {
+	private versionedNodeTypes: Map<string, VersionedTypes>;
```
Comment (tomi) [inline packages/@n8n/task-runner/src/node-types.ts:14]: What's the key here? Version id? Maybe we could add a comment or rename this to `nodeTypesByVersion`

File: packages/@n8n/task-runner/src/node-types.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+			versionedTypes.set(nt.name, versioned);
+		}
+
+		return versionedTypes;
+	}
+
+	// This isn't used in Workflow from what I can see
+	getByName(_nodeType: string): INodeType | IVersionedNodeType {
+		return undefined as unknown as INodeType | IVersionedNodeType;
```
Comment (tomi) [inline packages/@n8n/task-runner/src/node-types.ts:42]: Maybe we should throw instead to catch it here instead it breaking somewhere down the line?

File: packages/@n8n/task-runner/src/__tests__/node-types.test.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import type { INodeTypeDescription } from 'n8n-workflow';
+
+import { TaskRunnerNodeTypes } from '../node-types';
+
+const SINGLE_VERSIONED = { name: 'single-versioned', version: 1 };
+
+const SINGLE_UNVERISONED = { name: 'single-unversioned' };
```
Comment (tomi) [inline packages/@n8n/task-runner/src/__tests__/node-types.test.ts:7]: typo:
```suggestion
const SINGLE_UNVERSIONED = { name: 'single-unversioned' };
```

**Review by tomi — DISMISSED** [review]
Looks good 👏 Only couple small suggestion but those are not blockers

**Review by tomi — APPROVED** [review]
🚀

**Review by tomi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.65.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.65.0)

**Pull request merged**